napari/components/layerlist.py

```diff
@@ -57,7 +66,7 @@ def __init__(self, data=()) -> None:
         super().__init__(
             data=data,
             basetype=Layer,
-            lookup={str: lambda e: e.name},
+            lookup={str: get_name},
```

Comment:
I guess we could use:
```suggestion
            lookup={str: operator.attrgetter('name')},
```

But we wouldn't get type checking.

Reply:
I do not have a strong opinion here. I prefer not to use lambda because lambda stores local references.

Reply:
That's true for all closure, but closure tend to create issues with reference and leak objects. So I would prefer we avoid closures.